dashboard/app/main.go

```diff
@@ -1229,48 +1237,70 @@ func makeUIJob(job *Job, jobKey *db.Key, bug *Bug, crash *Crash, build *Build) *
 	return ui
 }
 
-func fetchErrorLogs(c context.Context) ([]byte, error) {
+func fetchErrorLogs() ([]byte, error) {
+	if !appengine.IsAppEngine() {
+		return nil, nil
+	}
+
 	const (
-		minLogLevel  = 3
-		maxLines     = 100
-		maxLineLen   = 1000
-		reportPeriod = 7 * 24 * time.Hour
+		maxLines   = 100
+		maxLineLen = 1000
 	)
-	q := &log.Query{
-		StartTime:     time.Now().Add(-reportPeriod),
-		AppLogs:       true,
-		ApplyMinLevel: true,
-		MinLevel:      minLogLevel,
+
+	ctx := context.Background()
+	projId := os.Getenv("GOOGLE_CLOUD_PROJECT")
+
+	adminClient, err := logadmin.NewClient(ctx, projId)
+	if err != nil {
+		return nil, errors.New("failed to create the logging client")
 	}
-	result := q.Run(c)
-	var lines []string
-	for i := 0; i < maxLines; i++ {
-		rec, err := result.Next()
-		if rec == nil {
+	defer adminClient.Close()
+
+	const name = "appengine.googleapis.com%2Frequest_log"
+
+	// default filter fetch last 24h
+	iter := adminClient.Entries(ctx,
+		logadmin.Filter(
+			fmt.Sprintf(`logName = "projects/%s/logs/%s" AND severity>="ERROR"`,
+				projId, name)),
+		logadmin.NewestFirst(),
+	)
+
+	var entries []*logging.Entry
+	for len(entries) < maxLines {
+		entry, err := iter.Next()
+		if err == iterator.Done {
 			break
 		}
 		if err != nil {
-			entry := fmt.Sprintf("ERROR FETCHING LOGS: %v\n", err)
-			lines = append(lines, entry)
-			break
+			return nil, err
 		}
-		for _, al := range rec.AppLogs {
-			if al.Level < minLogLevel {
+		entries = append(entries, entry)
+	}
+
+	var lines []string
+	for _, entry := range entries {
+		for _, logLine := range entry.Payload.(*proto.RequestLog).Line {
+			if logLine.GetSeverity() != ltype.LogSeverity_ERROR {
```

Comment:
We already filtered above by `AND severity>="ERROR"`. But that's entries vs lines?

Are there more severe levels like FATAL?

Reply:
I've seen the entries with 2 lines. First line was "INFO" level and second "ERROR".

Reply:
Higher levels are LogSeverity_CRITICAL, LogSeverity_ALERT and LogSeverity_EMERGENCY.
Good point. Lets use ">=" here instead of the "!=".

Reply:
Done.